spyder/api/mode.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © Spyder Project Contributors
+# Copyright © <2013-2016> <Colin Duquesnoy and others, see pyqode/AUTHORS.rst>
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""
+This module contains the editor extension API.
+Adapted from https://github.com/pyQode/pyqode.core/blob/master/pyqode/core/api/mode.py
+"""
+import logging
+
+
+def _logger():
+    return logging.getLogger(__name__)
+
+
+class Mode(object):
+    """
+    Base class for editor extensions. An extension is a "thing" that can be
```

Comment:
This should look like

    Base class for editor extensions.

    An extension is a "thing" that can be ...